core/cli/src/lib.rs

```diff
@@ -91,6 +93,17 @@ fn get_chain_key(cli: &SharedParams) -> String {
 	}
 }
 
+fn generate_node_name() -> String {
+	let node_name = Generator::with_naming(Name::Numbered).next().unwrap();
+	let cnt = node_name.chars().count();
+
+	if cnt >= MAX_NODE_NAME_LENGTH {
+		generate_node_name()
```

Comment:
instead of recursion, i would prefer a `loop` here that breaks when `node_name` is short enough.

Reply:
@rphmeier

Thank you for your advice.
Certainly, `loop` is simply and a better idea than recursion.
I refactored genarate_node_name function.

```
fn generate_node_name() -> String {
	let result = loop {
		let node_name = Generator::with_naming(Name::Numbered).next().unwrap();
		let count = node_name.chars().count();

		if count < MAX_NODE_NAME_LENGTH {
			break node_name
		}
	};
	
	result
}
```